### PR TITLE
soc: st: Add all stm32l010 missing SoCs

### PR DESCRIPTION
**Pull request by nandojve**
Add minimal configurations to enable the whole stm32l010xx SoCs

### CODE REVIEW

**Review by erwango — DISMISSED** [review]
Out of curiosity, do you an actual application using Zephyr on such a small target?

**Review by gautierg-st — DISMISSED** [review]

**Comment by nandojve** [comment]
> Out of curiosity, do you an actual application using Zephyr on such a small target?

Yes, see #63935 for more details. When I started this POC to try write a real application on it I was skeptical. I could ran a simple hello world with Zephyr, well tuned, in 7.5k flash and 568 bytes of ram. If I add `printk` it increase to around 9k flash. At end it is a _super loop_ application but you got all benefits from Zephyr to configure the hardware. When you compare with bare-metal it is easy to see that code not increase too much so for that application it still valid on my view.

**Review by FRASTM — DISMISSED** [review]

File: soc/arm/st_stm32/stm32l0/Kconfig.defconfig.stm32l010xx [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 # Copyright (c) 2021 Wouter Cappelle <wouter.cappelle@crodeon.com>
+# Copyright (c) 2023 OS Systems
 # SPDX-License-Identifier: Apache-2.0
 
-if SOC_STM32L010XB
+if SOC_STM32L010XX
 
 config SOC
-	default "stm32l010xb"
+	default "stm32l010xx"
```
Comment (erwango) [inline soc/arm/st_stm32/stm32l0/Kconfig.defconfig.stm32l010xx:10]: This won't work, CONFIG_SOC has to match defines available hre: https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32l0xx/soc/stm32l0xx.h#L61 (ie file names in https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32l0xx/soc/)

**Review by erwango — CHANGES_REQUESTED** [review]

File: soc/arm/st_stm32/stm32l0/Kconfig.defconfig.stm32l010xx (hunk 1 above)
Comment (nandojve) [reply soc/arm/st_stm32/stm32l0/Kconfig.defconfig.stm32l010xx:10]: Is the NUM_IRQS realy 32? The RM show from 0-29 IRQ vectors.
Can you confirm?

Comment (erwango) [reply soc/arm/st_stm32/stm32l0/Kconfig.defconfig.stm32l010xx:10]: Looking to https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32l0xx/soc/stm32l010x4.h#L96, it should be 30.

Comment (nandojve) [reply soc/arm/st_stm32/stm32l0/Kconfig.defconfig.stm32l010xx]: I send a separated patch as last commit to fix it.

**Review by otavio — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Comment by nandojve** [comment]
ping

File: dts/arm/st/l0/stm32l010X4.dtsi [hunk 2]
```diff
@@ -8,15 +8,15 @@
...
-
 	sram0: memory@20000000 {
 		reg = <0x20000000 DT_SIZE_K(2)>;
 	};
 
 	soc {
+		eeprom: eeprom@8080000{
+			reg = <0x08080000 128>;
+		};
+
```
Comment (gautierg-st) [inline dts/arm/st/l0/stm32l010X4.dtsi:19]: Please move this changes in your previous commit (when you add them)

File: dts/arm/st/l0/stm32l010X6.dtsi [hunk 3]
```diff
@@ -8,15 +8,15 @@
...
-
 	sram0: memory@20000000 {
 		reg = <0x20000000 DT_SIZE_K(8)>;
 	};
 
 	soc {
+		eeprom: eeprom@8080000{
+			reg = <0x08080000 256>;
+		};
+
```
Comment (gautierg-st) [inline dts/arm/st/l0/stm32l010X6.dtsi:19]: Same

File: dts/arm/st/l0/stm32l010X8.dtsi [hunk 4]
```diff
@@ -8,15 +8,15 @@
...
-
 	sram0: memory@20000000 {
 		reg = <0x20000000 DT_SIZE_K(8)>;
 	};
 
 	soc {
+		eeprom: eeprom@8080000{
+			reg = <0x08080000 256>;
+		};
+
```
Comment (gautierg-st) [inline dts/arm/st/l0/stm32l010X8.dtsi:19]: Same

**Review by gautierg-st — CHANGES_REQUESTED** [review]
Just a bit of reorganizing. Otherwise looks good

**Review by gautierg-st — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**